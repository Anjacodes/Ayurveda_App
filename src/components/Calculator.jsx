import { useState } from 'react';
import calculate from '../logic/calculate';
import { Button } from './Button';

export function Calculator() {
  const [state, setState] = useState({ total: '0', next: null, operation: null });

  const handleClick = (e) => {
    setState(calculate(state, e.target.innerText));
  };

  const printResults = () => {
    const { next, total, operation } = state;
    if (!next && !total && !operation) return '0';
    if (next && !total && !operation) return next;
    if (total && !next && operation) return total;
    if (total && next && operation) return next;
    return total;
  };

  return (
    <div className="mainContainer">
      <h2 className="title">Let&apos;s do some math!</h2>
      <div className="calcContainer">
        <p className="input">{printResults()}</p>
        <div className="gridContainer">
          <Button handleClick={handleClick} addClass="btn btn-grey" text="AC" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="+/-" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="%" />
          <Button handleClick={handleClick} addClass="btn-orange btn" text="รท" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="7" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="8" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="9" />
          <Button handleClick={handleClick} addClass="btn btn-orange" text="x" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="4" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="5" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="6" />
          <Button handleClick={handleClick} addClass="btn btn-orange" text="-" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="1" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="2" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="3" />
          <Button handleClick={handleClick} addClass="btn btn-orange" text="+" />
          <Button handleClick={handleClick} addClass="btn btn-large btn-grey" text="0" />
          <Button handleClick={handleClick} addClass="btn btn-grey" text="." />
          <Button handleClick={handleClick} addClass="btn btn-orange" text="=" />
        </div>
      </div>
    </div>
  );
}

export default Calculator;
