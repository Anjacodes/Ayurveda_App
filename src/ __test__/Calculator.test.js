import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator component', () => {
  it('Renders correctly calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
