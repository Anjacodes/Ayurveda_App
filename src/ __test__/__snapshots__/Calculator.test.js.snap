// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Renders correctly calculator component 1`] = `
<div
  className="mainContainer"
>
  <h2
    className="title"
  >
    Let's do some math!
  </h2>
  <div
    className="calcContainer"
  >
    <p
      className="input"
    >
      0
    </p>
    <div
      className="gridContainer"
    >
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-orange btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn btn-large btn-grey"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn btn-grey"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
